src/core/sorted_map.cc

```diff
@@ -123,15 +124,24 @@ unsigned char* zzlInsertAt(unsigned char* zl, unsigned char* eptr, sds ele, doub
     DCHECK_EQ(scorelen, strlen(scorebuf));
   }
 
+  // Argument parsing converts empty strings to default initialized string views.
+  // Such string views have a null data field, which if passed into lpAppend (via zzlInsertAt)
+  // results in the replace operation being applied on the listpack. In addition to being wrong, it
+  // also causes assertion failures. To circumvent this corner case we pass here a string view
+  // pointing to an empty string on the stack, which has a non-null data field.
+  if (ele.data() == nullptr) {
```

Comment:
Why don't we patch `lpAppend` ? Or if we want to enforce a contract, why not simply introduce a wrapper that does this check for us here and use it. That way, we save this "hack"? 

Not that it matters that much, it's a few lines

Reply:
we do not change code under redis directory and this function is the wrapper

Reply:
oh right you want to keep the diffs clean